app/assets/javascripts/components/overview/clone_course_button.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { cloneCourse } from '../../actions/course_creation_actions.js';
+import { initiateConfirm } from '../../actions/confirm_actions.js';
+import { updateCourse } from '../../actions/course_actions.js';
+
+CloneCourseButton.propTypes = {
+  courseId: PropTypes.number.isRequired,
+  cloneCourse: PropTypes.func.isRequired,
+  initiateConfirm: PropTypes.func.isRequired,
+  updateCourse: PropTypes.func.isRequired
+};
+
+export function CloneCourseButton(props) {
+  return (
+    <a onClick={onClickConfirmation(props)} className="button">
+      {I18n.t('courses.creator.clone_this')}
+    </a>
+  );
+}
+
+function onClickConfirmation(props) {
+  return () => {
+    const clone = () => {
+      props.cloneCourse(props.courseId).then(({ data }) => {
+        const course = data.course;
+        props.updateCourse(course);
+        window.location = `/courses/${course.slug}`;
```

Comment:
I'm not super happy about this. Ideally we'd use `withRouter` to give this component the `history` object so that we can use `history.push()`. However, doing so does not trigger the form to pop up for the cloned course. It looks like it happens on `UPDATE_COURSE`...